pkg/agent/multicluster/mc_route_controller.go

```diff
@@ -213,20 +242,202 @@ func (c *MCDefaultRouteController) processNextWorkItem() bool {
 	}
 	defer c.queue.Done(obj)
 
-	if k, ok := obj.(string); !ok {
+	key, ok := obj.(string)
+	if !ok {
 		c.queue.Forget(obj)
 		klog.InfoS("Expected string in work queue but got", "object", obj)
 		return true
-	} else if err := c.syncMCFlows(); err == nil {
-		c.queue.Forget(k)
+	}
+
+	syncFn := func() error {
+		if c.wireGuardConfig != nil {
+			if err := c.syncWireGuard(); err != nil {
+				return err
+			}
+		}
+		return c.syncMCFlows()
+	}
+	if err := syncFn(); err == nil {
+		c.queue.Forget(key)
 	} else {
 		// Put the item back on the workqueue to handle any transient errors.
-		c.queue.AddRateLimited(k)
-		klog.ErrorS(err, "Error syncing key, requeuing", "key", k)
+		c.queue.AddRateLimited(key)
+		klog.ErrorS(err, "Error syncing key, requeuing", "key", key)
 	}
 	return true
 }
 
+// syncWireGuard reconciles WireGuard configurations in following way:
+//  1. If the current Node is the Multi-cluster Gateway Node, controller will try to initialize corresponding WireGuard
+//     configuration and route on the host, then add all existing Gateway Nodes in other member clusters as WireGuard peers.
+//  2. If the current Node is not Multi-cluster Gateway Node, controller will try to clean up WireGuard configurations.
+//
+// Note: MCDefaultRouteController runs only one worker to process Gateway and ClusterInfoImport. So we do not need
+// any synchronization mechanism.
+func (c *MCDefaultRouteController) syncWireGuard() error {
+	gateway, err := c.getActiveGateway()
+	if err != nil {
+		return err
+	}
+
+	amIGateway := gateway != nil && gateway.Name == c.nodeConfig.Name
+	if c.wireGuardClient != nil && (!amIGateway || !c.wireGuardInitialized) {
+		if err := c.cleanUpWireGuard(); err != nil {
+			return err
+		}
+		c.wireGuardInitialized = false
+	}
+	if !amIGateway {
+		return nil
+	}
+	if !c.wireGuardInitialized {
+		var initErr error
+		defer func() {
+			if initErr != nil {
+				if err := c.cleanUpWireGuard(); err != nil {
+					klog.ErrorS(err, "Failed to clean up WireGuard")
+				}
+				c.wireGuardInitialized = false
+			}
+		}()
+		if initErr = c.initializeWireGuard(gateway); initErr != nil {
+			return initErr
+		}
+		c.wireGuardInitialized = true
+	}
+	ciImports, err := c.ciImportLister.List(labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	desireCIImport := sets.NewString()
+	var updateErr error
+	for _, ciImport := range ciImports {
+		desireCIImport.Insert(ciImport.Name)
+		if err = c.updateSingleWireGuardPeer(ciImport); err != nil {
+			klog.V(2).ErrorS(err, "Failed to update WireGuard peer", "ClusterInfoImport", klog.KObj(ciImport))
+			updateErr = err
+		}
+	}
+	if updateErr != nil {
+		return updateErr
+	}
+
+	// Check cache and existing ClusterInfoImport, clean up route and WireGuard peer of the
+	// removed ClusterInfoImport.
+	installCIImports := sets.StringKeySet(c.installedCIImports)
```

Comment:
```suggestion
	installedCIImports := sets.StringKeySet(c.installedCIImports)
```

Reply:
Do you need this new variable?
I suppose you can use range with `c.installedCIImports` directly?
```
for  name := range c.installedCIImports {
...
}
```

Reply:
Yes, the new variable could be removed. Done.